src/hardware/serialport/softmodem.cpp

```diff
@@ -355,6 +347,44 @@ char CSerialModem::GetChar(char * & scan) const {
 	return ch;
 }
 
+void CSerialModem::SetModemSpeed(const uint32_t cfg_val) {
+	modem_bps_config   = cfg_val;
+	const auto reported_bps = UpdateConnectString();
+
+	if (modem_bps_config) {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will report connection speed "
+		        "of %d bits per second",
+		        GetPortNumber(),
+		        reported_bps);
+	} else {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will synchronize reported speed "
+		        "to port baud rate",
+		        GetPortNumber());
+	}
+}
+
+uint32_t CSerialModem::UpdateConnectString() {
+	uint32_t connect_val;
+	if (modem_bps_config != 0) {
+		connect_val = clamp(modem_bps_config,
+		                    SERIAL_MIN_BAUDRATE,
+		                    SERIAL_MAX_BAUDRATE);
+		if (connect_val != modem_bps_config) {
+			LOG_WARNING("SERIAL: Port %" PRIu8 " modem's requested speed out of "
+			            "bounds, clamped to %d",
+			            GetPortNumber(),
+			            connect_val);
+		}
+	} else {
+		connect_val = GetPortBaudRate();
+	}
+
+	assert(connect_val >= SERIAL_MIN_BAUDRATE &&
+	       connect_val <= SERIAL_MAX_BAUDRATE);
+	safe_sprintf(connect_string, "CONNECT %d", connect_val);
+	return connect_val;
```

Comment:
These two functions serve to set `connect_string`. Because of this, suggest they be collapsed into a single simplified `UpdateConnectString()` that reads the `modem_bps_config` member variable.

This will also let you log the port rate in the case where the `modem_bps_config` is zero.

It will also let you assess if the `modem_bps_config` and port baud rate are compatible - for example, the `modem_bps_config` shouldn't be able to exceed the port rate (I think?).

Reply:
If you read the code closer you'll see that the former function is called specifically when the user updates modem speed whereas the latter is called individually whenever serial port config is updated. I've split them specifically so that logging is only done when user updates modem speed because updatePortConfig is called a lot and would cause log spam.

Reply:
I read close enough that they could be merged, but wasn't aware of the logging frequency. Sounds good.